google_cloud_storage/index.js

```diff
@@ -23,5 +28,42 @@ async function uploadFile(input) {
     await bucket.upload(filePath, {
 
     });
+
+    if (deleteOnUpload) {
+        fs.unlink(filePath, (err) => {
+            if (err) {
+                console.error("file was uploaded but did not get deleted from the local filesystem");
+            }
+            console.info("file deleted");
+        });
+    }
     return {"Success" : true };
+}
+
+async function downloadFile(input) {
+    apptree.validateInputs('Credential', 'FileName', 'ProjectId', 'Bucket', 'ServerFilePath');
+    let credential = input['Credential'];
+    let fileName = input['FileName'];
+    const bucketName = input['Bucket'];
+    const serverDir = input['ServerFilePath'];
+    // const deleteOnDownload = input['DeleteOnDownload'];
+
+    const storage = new Storage({
+        projectId: input['ProjectId'],
+        credentials: credential,
+    });
+    const bucket = storage.bucket(bucketName);
+    const exists = await bucket.exists();
+    if (!exists) {
+        console.error("Bucket does not exist");
+        process.exit(1);
```

Comment:
Use `throw` instead. It automatically sets the exit code correctly

`throw "Bucket does not exist"`

Reply:
updated